lib/commands.js

```diff
@@ -37,6 +57,56 @@ export default class Commands {
   toggleActiveEditor() {
     this.emitter.emit('should-toggle-active-editor')
   }
+  showDebug(standardLinters: Array<Linter>, indieLinters: Array<IndieDelegate>, uiProviders: Array<UI>) {
+    if (!manifest) {
+      manifest = require('../package.json')
+    }
+    if (!Helpers) {
+      Helpers = require('./helpers')
+    }
+
+    const textEditor = atom.workspace.getActiveTextEditor()
+    const textEditorScopes = Helpers.getEditorCursorScopes(textEditor)
+    const sortedLinters = standardLinters.slice().sort(sortByName)
+    const sortedIndieLinters = indieLinters.slice().sort(sortByName)
+    const sortedUIProviders = uiProviders.slice().sort(sortByName)
+
+    const indieLinterNames = sortedIndieLinters
+      .map(formatItem).join('\n')
+    const standardLinterNames = sortedLinters
+      .map(formatItem).join('\n')
+    const matchingStandardLinters = sortedLinters
+      .filter(linter => Helpers.shouldTriggerLinter(linter, false, textEditorScopes))
+      .map(formatItem).join('\n')
+    const humanizedScopes = textEditorScopes
+      .map(formatItem).join('\n')
+    const uiProviderNames = sortedUIProviders
+      .map(formatItem).join('\n')
+
+    const ignoreGlob = atom.config.get('linter.ignoreGlob')
+    const ignoreVCSIgnoredPaths = atom.config.get('core.excludeVcsIgnoredPaths')
+    const disabledLinters = atom.config.get('linter.disabledProviders')
+      .map(formatItem).join('\n')
+    const filePathIgnored = Helpers.isPathIgnored(textEditor.getPath(), ignoreGlob, ignoreVCSIgnoredPaths)
+
+    atom.notifications.addInfo('Linter Debug Info', {
+      detail: [
+        `Platform: ${process.platform}`,
+        `Atom Version: ${atom.getVersion()}`,
+        `Linter Version: ${manifest.version}`,
+        `Ignore Glob: ${ignoreGlob}`,
+        `Ignore VCS Ignored Paths: ${ignoreVCSIgnoredPaths}`,
+        `Opened File is ignored: ${filePathIgnored ? 'Yes' : 'No'}`,
+        `UI Providers: \n${uiProviderNames}`,
+        `Indie Linter Providers: \n${indieLinterNames}`,
+        `Matching Linter Providers: \n${matchingStandardLinters}`,
```

Comment:
Here's the order I think would be the easiest to quickly review when viewing posted issues:
* Platform
* Atom Version
* Linter Version
* Opened File is ignored
* Matching Linter Providers
* Disabled Linter Providers
* Standard Linter Providers
* Indie Linter Providers
* UI Providers
* Ignore Glob
* Ignore VCS Ignored Paths
* Current File scopes

This will put the information in order of relevance to most issues.